packages/ember-glimmer/tests/utils/shared-conditional-tests.js

```diff
@@ -374,6 +384,53 @@ export class SharedConditionalsTest extends AbstractConditionalsTest {
     this.assertText('T1F2');
   }
 
+  ['@test it maintains DOM stability when condition changes from a truthy to a different truthy value']() {
+    this.renderValues(true);
+
+    this.assertText('T1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', 'hello'));
+
+    this.assertText('T1');
+
+    this.assertInvariants();
+  }
+
+  ['@test it maintains DOM stability when condition changes from a falsy to a different falsy value']() {
+    this.renderValues(false);
+
+    this.assertText('F1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', ''));
+
+    this.assertText('F1');
+
+    this.assertInvariants();
+  }
+}
+
+export class SharedHelperConditionalsTest extends SharedConditionalsTest {
+  renderValues(...values) {
+    let templates = [];
+    let context = {};
+
+    for (let i = 1; i <= values.length; i++) {
+      templates.push(this.templateFor({ cond: `cond${i}`, truthy: `t${i}`, falsy: `f${i}` }));
+      context[`t${i}`] = `T${i}`;
+      context[`f${i}`] = `F${i}`;
+      context[`cond${i}`] = values[i - 1];
+    }
+
+    let wrappedTemplate = this.wrapperFor(templates);
+    this.render(wrappedTemplate, context);
+  }
+}
+
+export class SharedSyntaxConditionalsTest extends SharedConditionalsTest {
   ['@htmlbars it does not update when the unbound helper is used']() {
```

Comment:
Moved to `SharedSyntaxConditionalsTest` as this template isn't compatible with `SharedHelperConditionalsTest`. (Having said that, I'm going to reimplement a `SharedHelperConditionalsTest` version of it now)